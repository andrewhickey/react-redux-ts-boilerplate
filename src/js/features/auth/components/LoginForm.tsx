import { Button, Classes, InputGroup } from "@blueprintjs/core"
import * as React from "react"
import { Field, GenericField, reduxForm, WrappedFieldProps } from "redux-form"

export const UsernameInput: React.StatelessComponent<WrappedFieldProps> = ({
  input,
  meta,
}: WrappedFieldProps) => (
  <div className="pt-form-group pt-intent-danger">
    <InputGroup
      placeholder="Username"
      id="username-input"
      className={Classes.LARGE}
      leftIconName="person"
      {...input}
    />
    {meta.error && <div className="pt-form-helper-text">meta.error</div>}
  </div>
)

export const PasswordInput: React.StatelessComponent<WrappedFieldProps> = ({
  input,
  meta,
}: WrappedFieldProps) => (
  <div className="pt-form-group pt-intent-danger">
    <InputGroup
      placeholder="Password"
      id="password-input"
      className={Classes.LARGE}
      leftIconName="lock"
      type="password"
      {...input}
    />
    {meta.error && <div className="pt-form-helper-text">meta.error</div>}
  </div>
)

const WrapperField = Field as new () => GenericField<{}>

export const LoginFormComponent = () => (
  <form>
    <div className="pt-control-group pt-vertical">
      <WrapperField name="username" component={UsernameInput} />
      <WrapperField name="password" component={PasswordInput} />
      <Button id="login-button" className={Classes.LARGE} iconName="log-in" />
    </div>
  </form>
)

export default reduxForm({ form: "login" })(LoginFormComponent)
