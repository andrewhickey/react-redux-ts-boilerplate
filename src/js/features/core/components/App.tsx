import * as React from 'react'

export default () => <div id="appContainer" />
