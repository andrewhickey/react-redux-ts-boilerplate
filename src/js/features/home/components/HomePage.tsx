import * as React from "react"
import { withProps } from "recompose"

export interface IHomeProps {
  name: string
  number: number
}

export const HomePageComponent = ({ name }: IHomeProps) => (
  <div>Hello dog {name}</div>
)

export const homeWrapper1 = withProps({ name: "dog" })
export const homeWrapper2 = withProps({ number: 2 })

export default homeWrapper2(homeWrapper1(HomePageComponent))
