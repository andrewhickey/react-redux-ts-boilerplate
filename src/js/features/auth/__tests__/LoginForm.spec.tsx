import { shallow } from "enzyme"
import * as React from "react"
import { LoginFormComponent } from "../components/LoginForm"

describe("LoginFormComponent", () => {
  it("Matches the snapshot", () => {
    const wrapper = shallow(<LoginFormComponent />)
    expect(wrapper).toMatchSnapshot()
  })
  it("Has a button to submit the login", () => {
    const wrapper = shallow(<LoginFormComponent />)
    expect(wrapper.find("#login-button")).toBePresent()
  })
})
