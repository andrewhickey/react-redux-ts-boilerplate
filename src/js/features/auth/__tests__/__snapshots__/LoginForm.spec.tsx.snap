// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormComponent Matches the snapshot 1`] = `
<form>
  <div
    className="pt-control-group pt-vertical"
  >
    <Field
      component={[Function]}
      name="username"
    />
    <Field
      component={[Function]}
      name="password"
    />
    <Blueprint.Button
      className="pt-large"
      iconName="log-in"
      id="login-button"
    />
  </div>
</form>
`;
